File: pegasusX/apps/driver-app-android/app/src/main/java/com/pegasusx/driver/ui/screens/home/components/VehicleInfoCard.kt
package com.pegasusx.driver.ui.screens.home.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.LocalShipping
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.pegasusx.driver.ui.components.PegasusCard
import com.pegasusx.driver.ui.components.StatusPill
import com.pegasusx.driver.ui.theme.LocalPegasusColors
import com.pegasusx.driver.ui.theme.PegasusSpacing

@Composable
fun VehicleInfoCard(
    truckId: String,
    licensePlate: String,
    vehicleClass: String,
    maxVolumeVU: Double
) {
    val lab = LocalPegasusColors.current
    PegasusCard {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(PegasusSpacing.s16),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(14.dp)
        ) {
            Box(
                modifier = Modifier
                    .size(48.dp)
                    .clip(RoundedCornerShape(12.dp))
                    .background(lab.separator),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = Icons.Default.LocalShipping,
                    contentDescription = null,
                    tint = lab.fg,
                    modifier = Modifier.size(24.dp)
                )
            }

            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = truckId,
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    color = lab.fg
                )
                val subtitle = buildString {
                    append(licensePlate)
                    if (vehicleClass.isNotBlank()) {
                        append(" · $vehicleClass · ${maxVolumeVU.toInt()} VU")
                    }
                }
                Text(
                    text = subtitle,
                    style = MaterialTheme.typography.bodyMedium,
                    fontWeight = FontWeight.Medium,
                    fontFamily = FontFamily.Monospace,
                    color = lab.fgTertiary
                )
            }

            StatusPill(label = "ASSIGNED", color = lab.success)
        }
    }
}

File: pegasusX/apps/driver-app-ios/driverappios/driverappios/Views/Components/Home/VehicleInfoCard.swift
import SwiftUI

struct VehicleInfoCard: View {
    var vm: FleetViewModel

    var body: some View {
        HStack(spacing: 14) {
            Image(systemName: "truck.box.fill")
                .font(.system(size: 22))
                .foregroundStyle(LabTheme.fg)
                .frame(width: 44, height: 44)
                .background(LabTheme.separator)
                .clipShape(.rect(cornerRadius: 12))

            VStack(alignment: .leading, spacing: 4) {
                Text(vm.truckId)
                    .font(.system(size: 15, weight: .bold))
                    .foregroundStyle(LabTheme.fg)
                HStack(spacing: 6) {
                    Text(vm.licensePlate)
                        .font(.system(size: 12, weight: .medium, design: .monospaced))
                        .foregroundStyle(LabTheme.fgTertiary)
                    if vm.vehicleClass != "—" {
                        Text("•")
                            .foregroundStyle(LabTheme.fgTertiary)
                        Text(L10n.format("mobile_driver.ui.vehicleclass_maxvolumevu_vu", "\(vm.vehicleClass)", "\(Int(vm.maxVolumeVU))"))
                            .font(.system(size: 12, weight: .medium, design: .monospaced))
                            .foregroundStyle(LabTheme.fgTertiary)
                    }
                }
            }

            Spacer()

            DriverStatusBadge(text: "ASSIGNED", tint: LabTheme.success)
        }
        .padding(LabTheme.s16)
        .labCard()
    }
}
